File: padlock/src/main/java/com/pyamsoft/padlock/app/service/RecheckService.java
/*
 * Copyright 2016 Peter Kenji Yamanaka
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.pyamsoft.padlock.app.service;

import android.app.IntentService;
import android.content.Intent;
import android.support.annotation.NonNull;
import timber.log.Timber;

public class RecheckService extends IntentService {

  @NonNull public static final String EXTRA_PACKAGE_NAME = "extra_package_name";
  @NonNull public static final String EXTRA_CLASS_NAME = "extra_class_name";

  public RecheckService() {
    super(RecheckService.class.getName());
  }

  @Override protected void onHandleIntent(Intent intent) {
    if (intent == null) {
      Timber.e("Intent is NULL");
      return;
    }

    if (!intent.hasExtra(EXTRA_PACKAGE_NAME)) {
      Timber.e("No package name passed");
      return;
    }

    if (!intent.hasExtra(EXTRA_CLASS_NAME)) {
      Timber.e("No class name passed");
      return;
    }

    final String packageName = intent.getStringExtra(EXTRA_PACKAGE_NAME);
    final String className = intent.getStringExtra(EXTRA_CLASS_NAME);
    try {
      PadLockService.recheck(packageName, className);
    } catch (NullPointerException e) {
      Timber.e(e, "ERROR");
    }
  }
}

File: padlock/src/main/java/com/pyamsoft/padlock/service/RecheckService.kt
/*
 *     Copyright (C) 2017 Peter Kenji Yamanaka
 *
 *     This program is free software; you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation; either version 2 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License along
 *     with this program; if not, write to the Free Software Foundation, Inc.,
 *     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package com.pyamsoft.padlock.service

import android.app.IntentService
import android.content.Intent
import com.pyamsoft.padlock.Injector
import com.pyamsoft.padlock.PadLockComponent
import com.pyamsoft.padlock.lock.Recheck
import timber.log.Timber
import javax.inject.Inject

class RecheckService : IntentService(RecheckService::class.java.name) {

  @field:Inject internal lateinit var recheckBus: RecheckPublisher

  override fun onCreate() {
    super.onCreate()
    Injector.obtain<PadLockComponent>(applicationContext).inject(this)
  }

  override fun onHandleIntent(intent: Intent?) {
    if (intent == null) {
      Timber.e("Intent is NULL")
      return
    }

    if (!intent.hasExtra(Recheck.EXTRA_PACKAGE_NAME)) {
      Timber.e("No package name passed")
      return
    }

    if (!intent.hasExtra(Recheck.EXTRA_CLASS_NAME)) {
      Timber.e("No class name passed")
      return
    }

    val packageName = intent.getStringExtra(Recheck.EXTRA_PACKAGE_NAME)
    val className = intent.getStringExtra(Recheck.EXTRA_CLASS_NAME)
    if (packageName.isNotEmpty() && className.isNotEmpty()) {
      Timber.d("Recheck was requested for: %s, %s", packageName, className)
      recheckBus.publish(RecheckEvent.create(packageName, className))
    }
  }
}
